LibSWBF2.NET/Wrappers/Level.cs

```diff
@@ -96,5 +112,19 @@ public Model GetModel(string modelName)
             Children.Add(new WeakReference<NativeWrapper>(model));
             return model;
         }
+
+
+        public bool GetTexture(string name, out int width, out int height, out byte[] texBytes)
+        {
+            texBytes = null;
+            bool result = APIWrapper.Level_GetTextureData(NativeInstance, name, out IntPtr bytesRaw, out width, out height);
+            if (result)
+            {
+                texBytes = new byte[width * height * 4];
+                Marshal.Copy(bytesRaw, texBytes, 0, width * height * 4);
+            }
+
+            return result;
```

Comment:
Removed all ```FreeHGlobal``` calls from the C# API, the memory is now managed through static local variables in the ```API.cpp``` getters and the ```Terrain``` wrapper.